frontend/src/lib/notify/index.ts

```diff
@@ -38,6 +42,10 @@ export class NotificationService {
     this._notifications.set([]);
   }
 
+  private addPersistentNotification(notification: Notification): void {
+    this._notifications.update((currentNotifications) => [...currentNotifications, notification]);
+  }
+
   private addNotification(notification: Notification): void {
     this._notifications.update((currentNotifications) => [...currentNotifications, notification]);
     setTimeout(() => {
```

Comment:
Right here, I'd suggest wrapping an `if` block around this setTimeout so that it only sets a timeout if `notification.duration > 0`. That way making a notification persistent is as easy as `notifyWarning(message, 0)`. We could even add a `Persistent = 0` value to the Duration enum so that you can write that as `notifyWarning(message, Duration.Persistent)`.